torch/testing/_internal/distributed/distributed_test.py

```diff
@@ -1036,6 +1036,80 @@ def test_periodic_model_averager(self):
                         # No model averaging, so the parameters are not updated.
                         self.assertEqual(param.data, tensor)
 
+        @skip_if_lt_x_gpu(2)
+        def test_periodic_model_averager_param_group_fast(self):
+            rank = dist.get_rank()
+            world_size = dist.get_world_size()
+            rank_to_GPU = init_multigpu_helper(world_size, BACKEND)
+            device_id = rank_to_GPU[rank][0]
+
+            model = nn.Linear(1, 5, bias=False).cuda(device_id)
+            param = next(model.parameters())
+            opt = torch.optim.SGD(model.parameters(), lr=0.1)
+
+            period = 4
+            for warmup_steps in [12, 13, 14, 15]:
+                averager = averagers.PeriodicModelAverager(period=period, warmup_steps=warmup_steps)
+                for step in range(0, 20):
+                    # Reset the parameters at every step.
+                    for param_group in opt.param_groups:
+                        for params in param_group["params"]:
+                            # mock grad
+                            params.grad = torch.ones_like(param.data) * rank
```

Comment:
Compared with the existing test case `test_periodic_model_averager`, I think you are testing `param_groups` as an alternate arg here.

If we move this test to `LocalSGDOptimizer`, will `_test_post_localSGD_optimizer_parity` be sufficient?

Reply:
_test_post_localSGD_optimizer_parity is only to compare the origin optim  update params with LocalSGDOptimizer  update params.
but not test the validation of  update param.
but i donot find a simple way to mock the update param.
so i tried to mock the usage of LocalSGDOptimizer.
so test_periodic_model_averager_param_group_fast is to test LocalSGDOptimizer update param validation  although it is familar with test_periodic_model_averager.
Do you have any idea how to mock the update param validation in _test_post_localSGD_optimizer_parity?   
i do not find a simple way so i create a new unit test

Reply:
See the new comment: https://github.com/pytorch/pytorch/pull/74894#discussion_r841193929

Now I agree that better to handle parameter groups inside `average_parameters`, then it makes sense to create another test case like this. Please ignore my previous comment in this thread.